src/common/snippets/src/op/subgraph.cpp

```diff
@@ -470,70 +480,38 @@ bool Subgraph::check_broadcast(const std::shared_ptr<const ov::Node>& node) noex
 }
 
 IShapeInferSnippets::Result Subgraph::shape_infer(const std::vector<VectorDimsRef>& input_shapes) {
-    if (!m_shape_infer && !m_linear_ir) {
-        OPENVINO_ASSERT(body_ptr(), "Can't create shape infer for Subgraph with an empty body");
-        m_shape_infer = std::make_shared<NgraphShapeInfer>(body_ptr());
-    } else if (!std::dynamic_pointer_cast<LIRShapeInfer>(m_shape_infer) && m_linear_ir) {
-        m_shape_infer = std::make_shared<LIRShapeInfer>(m_linear_ir);
-    }
+    OPENVINO_ASSERT(m_shape_infer && body_ptr(), "Attempt to call shape_infer when it's not initialized or on empty body");
```

Comment:
You know, I think we don't even need to check `body_ptr()`, because we do it in the `NgraphShapeInfer` constructor. Moreover, the whole shape infer is performed on this saved body ptr, so I think this check is not relevant anymore.

Reply:
Agree with you. Thanks for the explanation